diskqueue.go

```diff
@@ -99,14 +102,15 @@ type diskQueue struct {
 	exitChan          chan int
 	exitSyncChan      chan int
 
-	logf AppLogFunc
+	logf          AppLogFunc
+	emptyCallback EmptyCallback
 }
 
 // New instantiates an instance of diskQueue, retrieving metadata
 // from the filesystem and starting the read ahead goroutine
 func New(name string, dataPath string, maxBytesPerFile int64,
 	minMsgSize int32, maxMsgSize int32,
-	syncEvery int64, syncTimeout time.Duration, logf AppLogFunc) Interface {
+	syncEvery int64, syncTimeout time.Duration, logf AppLogFunc, cb EmptyCallback) Interface {
```

Comment:
I have quibbles with the naming here ... the `type EmptyCallback func()` initially sounds to me like it is describing the callback taking no arguments and returning nothing. I lean towards just `queueEmptyCB func()` (no extra `type`).

It's a bit annoying that we have to break compatibility when adding optional arguments here. Already did that for `logf` though, and not sure want to redesign with a DiskQueueOptions struct or similar, right now.